### PR TITLE
feat(util & Client): add createOAuthURL

### LINKED ISSUE #226: [Feature Request] Invite link utility function
**What do you want to request?**
A utility function to create bot/application invites. 
It  would work the same as discord.py's [discord.utils.oauth_url](https://discordpy.readthedocs.io/en/stable/api.html#discord.utils.oauth_url) function. 

It would be used to easily create bot invites or oauth URLs to get a user's token for connections or guilds or whatever.

### PR DESCRIPTION
**Pull request by DjDeveloperr**
fix #226

### CODE REVIEW

File: src/utils/oauthURL.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+              options.permissions instanceof Permissions
+            ? options.permissions.bitfield.toString()
+            : '0'
+        }`
+      : ''
+  }${
+    options.redirectURI !== undefined
+      ? `&redirct_uri=${options.redirectURI}`
+      : ''
+  }`
```
Comment (Helloyunho) [inline src/utils/oauthURL.ts:40]: maybe split some codes to several variables
too much things are in one template literal

**Review by Helloyunho — CHANGES_REQUESTED** [review]

File: src/utils/oauthURL.ts (hunk 1 above)
Comment (DjDeveloperr) [reply src/utils/oauthURL.ts]: done

Comment (DjDeveloperr) [reply src/utils/oauthURL.ts]: i miss that ternary 🥺

**Review by Helloyunho — APPROVED** [review]
lgtm :imwave:

**Pull request merged**